xbmc/FileItem.cpp

```diff
@@ -490,8 +476,7 @@ void CFileItem::Reset()
   m_epgInfoTag.reset();
   delete m_pvrChannelInfoTag;
   m_pvrChannelInfoTag=NULL;
-  delete m_pvrRecordingInfoTag;
-  m_pvrRecordingInfoTag=NULL;
+  m_pvrRecordingInfoTag.reset();
```

Comment:
Wait, I'm not quite sure whether I'm getting your point. shared_ptr::reset() does not touch the data of the object held by the shared_ptr instance, it (basically) just nulls its internal ptr member, right? So calling FileItem::Reset() which calls several .reset() then should not harm the objects held before the reset(). Or did I get some very basic stuff wrong?